lib/weka/core/instances.rb

```diff
@@ -266,6 +307,50 @@ def map_attribute_type(type)
         return -1 unless Attribute::TYPES.include?(type.downcase.to_sym)
         Attribute.const_get(type.upcase)
       end
+
+      # Convert a hash whose keys are attribute names and values are attribute
+      #   values into an array containing attribute values in the order
+      #   of the Instances attributes.
+      #
+      # @param [Hash] attribute_values a hash whose keys are attribute
+      #   names and values are attribute values.
+      #
+      # @return [Array] an array containing attribute values in the
+      #   correct order
+      def attribute_values_from_hash(attribute_values)
+        names = attribute_names(true).map!(&:to_sym)
+        names.inject([]) do |values, attribute_name|
+          values << attribute_values[attribute_name]
+        end
+      end
+
+      # Convert an array of attribute values in the same order as Instances
+      #   attributes into a hash whose keys are attribute names and values
+      #   are corresponding attribute values.
+      #
+      # @param [Array] attribute_values an array containing the attribute
+      #   values
+      #
+      # @return [Hash] a hash as described above
+      def attribute_values_to_hash(attribute_values)
+        names = attribute_names(true).map!(&:to_sym)
+        names.each_with_object({}).with_index do |(attr_name, hash), index|
+          hash[attr_name] = attribute_values[index]
+        end
```

Comment:
Is `values` instead of `attribute_values` and `name` for `attr_name` enough? Just trying to make it as concise as possible.

Maybe using inject is slightly more readable. What about sth. like this?:

```ruby
names.each_with_index.inject({}) do |hash, (name, index)|
  hash.update(name => values[index])
end
```